sources/api/netdog/src/cli/update_responder.rs

```diff
@@ -0,0 +1,130 @@
+#[cfg(net_backend = "systemd-networkd")]
+use super::{error, primary_interface_name, Result};
+#[cfg(net_backend = "systemd-networkd")]
+use crate::dns::DnsSettings;
+#[cfg(net_backend = "systemd-networkd")]
+use crate::networkd_status::{get_link_status, NetworkdStatus};
+#[cfg(net_backend = "systemd-networkd")]
+use crate::{CURRENT_IP, PRIMARY_SYSCTL_CONF, SYSCTL_MARKER_FILE, SYSTEMD_SYSCTL};
+#[cfg(net_backend = "systemd-networkd")]
+use argh::FromArgs;
+#[cfg(net_backend = "systemd-networkd")]
+use snafu::{ensure, ResultExt};
+#[cfg(net_backend = "systemd-networkd")]
+use std::fmt::Write;
+#[cfg(net_backend = "systemd-networkd")]
+use std::fs;
+#[cfg(net_backend = "systemd-networkd")]
+use std::net::IpAddr;
+#[cfg(net_backend = "systemd-networkd")]
+use std::path::Path;
+#[cfg(net_backend = "systemd-networkd")]
+use std::process::Command;
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "update-responder")]
+#[cfg(net_backend = "systemd-networkd")]
+/// Updates the various files needed when responding to events
+pub(crate) struct UpdateResponderArgs {}
+
+/// Updates the various files needed when responding to events
+#[cfg(net_backend = "systemd-networkd")]
+pub(crate) fn run() -> Result<()> {
+    let primary_interface = primary_interface_name()?;
+
+    let primary_link_status = get_link_status(primary_interface.clone()).unwrap();
+
+    // Write out current IP
+    let primary_ip = &primary_link_status.primary_address().unwrap();
+    write_current_ip(primary_ip)?;
+
+    // Write out resolv.conf
+    write_resolv_conf(&primary_link_status)?;
+
+    // If we haven't already, set and apply default sysctls for the primary network
+    // interface
+    if !Path::exists(Path::new(PRIMARY_SYSCTL_CONF)) {
+        write_interface_sysctl(primary_interface, PRIMARY_SYSCTL_CONF)?;
+    };
+
+    // Execute `systemd-sysctl` with our configuration file to set the sysctls
+    if !Path::exists(Path::new(SYSCTL_MARKER_FILE)) {
```

Comment:
I got some interesting issues coming back and forth from some isolated IPv6 testing but its a little tough chasing so many one-off issues. We may need to wait to confirm this until we are generating the config rather than hardcoding it since my testing may not be representative of the true state we will have.